bot/exts/moderation/voice_gate.py

```diff
@@ -0,0 +1,154 @@
+    @in_whitelist(channels=(Channels.voice_gate,), redirect=None)
+    async def voice_verify(self, ctx: Context, *_) -> None:
+        """
+        Apply to be able to use voice within the Discord server.
+
+        In order to use voice you must meet all three of the following criteria:
+        - You must have over a certain number of messages within the Discord server
+        - You must have accepted our rules over a certain number of days ago
+        - You must not be actively banned from using our voice channels
+        """
+        # Send this as first thing in order to return after sending DM
+        await ctx.send(f"{ctx.author.mention}, check your DMs.")
+
+        try:
+            data = await self.bot.api_client.get(f"bot/users/{ctx.author.id}/metricity_data")
+        except ResponseCodeError as e:
+            if e.status == 404:
+                embed = discord.Embed(
+                    title="Not found",
+                    description=(
+                        "We were unable to find user data for you. "
+                        "Please try again shortly, "
+                        "if this problem persists please contact the server staff through Modmail.",
+                    ),
+                    color=Colour.red()
+                )
+                log.info(f"Unable to find Metricity data about {ctx.author} ({ctx.author.id})")
+            else:
+                embed = discord.Embed(
+                    title="Unexpected response",
+                    description=(
+                        "We encountered an error while attempting to find data for your user. "
+                        "Please try again and let us know if the problem persists."
+                    ),
+                    color=Colour.red()
+                )
+                log.warning(f"Got response code {e.status} while trying to get {ctx.author.id} Metricity data.")
+
+            await ctx.author.send(embed=embed)
+            return
+
+        # Pre-parse this for better code style
+        if data["verified_at"] is not None:
+            data["verified_at"] = parser.isoparse(data["verified_at"])
+        else:
+            data["verified_at"] = datetime.utcnow() - timedelta(days=3)
+
+        checks = {
+            "verified_at": data["verified_at"] > datetime.utcnow() - timedelta(days=GateConf.minimum_days_verified),
+            "total_messages": data["total_messages"] < GateConf.minimum_messages,
+            "voice_banned": data["voice_banned"]
+        }
+        failed = any(checks.values())
+        failed_reasons = [MESSAGE_FIELD_MAP[key] for key, value in checks.items() if value is True]
+        [self.bot.stats.incr(f"voice_gate.failed.{key}") for key, value in checks.items() if value is True]
+
+        if failed:
+            embed = discord.Embed(
+                title="Voice Gate not passed",
+                description=FAILED_MESSAGE.format(reasons="\n".join(f'- You {reason}.' for reason in failed_reasons)),
```

Comment:
Let's use bullet points here actually.

```suggestion
                description=FAILED_MESSAGE.format(reasons="\n".join(f'• You {reason}.' for reason in failed_reasons)),
```